File: Queue (Python)/queue_.py
class Queue_:
    """Provides the basic functionalities of a queue data structure."""

    # Dunder methods

    def __init__(self) -> None:

        # The main queue list
        self._queue: list = []
        # The length of the queue
        self._last: int = -1
        # The index 0 constant
        self.__first: int = 0

    # Properties

    @property
    def queue(self) -> list:
        """Returns the queue list itself."""
        return self._queue

    @property
    def last(self) -> int:
        """Returns the index at which the last element sits. -1 means the queue is empty."""
        return self._queue[self._last]

    @property
    def first(self) -> int:
        """Returns the index at which the last element sits. -1 means the queue is empty."""
        return self._queue[0]

    # Instance methods

    def enqueue(self, value) -> None:
        """Appends the given value to the end of the queue."""
        self._last += 1
        self._queue.append(value)

    def dequeue(self) -> object:
        """Removes and returns the first element in the queue."""
        if self._last < self.__first:
            raise KeyError("The queue is empty.")
        self._last -= 1
        return self._queue.pop(0)

File: Queue (C++)/Queue.cpp
#include "Queue.h"
#include <iostream>

Queue::Queue(int length)
{
    m_length = length;
    m_queue = new int[length]{};
}

Queue::~Queue()
{
    delete[] m_queue;
}

void Queue::enqueue(int value)
{
    m_last++;
    if (m_last >= m_length)
    {
        throw "The queue is full.";
        m_last--;
    }
    else
    {
        m_queue[m_last] = value;
    }
}

int Queue::dequeue()
{
    if (m_last < m_first)
    {
        throw "The queue is empty.";
    }
    else
    {
        int element = m_queue[0];
        for (int i{0}; i < m_last; i++)
        {
            m_queue[i] = m_queue[i + 1];
        }
        m_last--;
        return element;
    }
}

int Queue::last()
{
    if (m_last < m_first)
    {
        throw "The queue is empty.";
    }
    else
    {
        return m_queue[m_last];
    }
}

int Queue::first()
{
    if (m_last < m_first)
    {
        throw "The queue is empty.";
    }
    else
    {
        return m_queue[m_first];
    }
}

void Queue::print()
{
    std::cout << "{ ";
    for (int i{0}; i <= m_last; i++)
    {
        std::cout << m_queue[i] << " ";
    }
    std::cout << "}" << std::endl;
}